MailCore/Models/Draft.swift

```diff
@@ -482,7 +185,120 @@ public class Draft: Object, Decodable, Identifiable, AbstractDraft {
         self.action = action
     }
 
-    public func asUnmanaged() -> UnmanagedDraft {
-        return .toUnmanaged(managedDraft: self)
+    public static func mailTo(subject: String?,
+                              body: String?,
+                              to: [Recipient], cc: [Recipient], bcc: [Recipient]) -> Draft {
+        return Draft(subject: subject ?? "",
+                     body: body ?? "",
+                     to: to,
+                     cc: cc,
+                     bcc: bcc)
+    }
+
+    public static func writing(to recipient: Recipient) -> Draft {
+        return Draft(to: [recipient.detached()])
+    }
+
+    public static func replying(to message: Message, mode: ReplyMode, localDraftUUID: String) -> Draft {
+        var subject = "\(message.formattedSubject)"
+        let quote: String
+        var attachments: [Attachment] = []
+        switch mode {
+        case .reply, .replyAll:
+            if !subject.starts(with: "Re: ") {
+                subject = "Re: \(subject)"
+            }
+            quote = Constants.replyQuote(message: message)
+        case let .forward(attachmentsToForward):
+            if !subject.starts(with: "Fwd: ") {
+                subject = "Fwd: \(subject)"
+            }
+            quote = Constants.forwardQuote(message: message)
+            attachments = attachmentsToForward
+        }
+
+        var to: [Recipient] = []
+        var cc: [Recipient] = []
+
+        if mode.isReply {
+            let userEmail = AccountManager.instance.currentMailboxManager?.mailbox.email ?? ""
+            let cleanedFrom = Array(message.from.detached()).filter { $0.email != userEmail }
+            let cleanedTo = Array(message.to.detached()).filter { $0.email != userEmail }
+            let cleanedReplyTo = Array(message.replyTo.detached()).filter { $0.email != userEmail }
+            let cleanedCc = Array(message.cc.detached()).filter { $0.email != userEmail }
+
+            to = cleanedReplyTo.isEmpty ? cleanedFrom : cleanedReplyTo
```

Comment:
It looks like an unrolled loop, probably easier to review than a `while(to.isEmpty) { }.`
I can't think of something easier to read ATM.

Reply:
discussed